src/main/java/com/mcmoddev/lib/container/MMDContainer.java

```diff
@@ -0,0 +1,190 @@
+package com.mcmoddev.lib.container;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import com.google.common.collect.Lists;
+import com.mcmoddev.lib.MMDLib;
+import com.mcmoddev.lib.container.gui.GuiContext;
+import com.mcmoddev.lib.container.widget.IContextualWidget;
+import com.mcmoddev.lib.container.widget.IProxyWidget;
+import com.mcmoddev.lib.container.widget.IWidget;
+import com.mcmoddev.lib.network.MMDPackages;
+import com.mcmoddev.lib.network.NBTBasedPlayerMessage;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.entity.player.EntityPlayerMP;
+import net.minecraft.inventory.Container;
+import net.minecraft.inventory.Slot;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraftforge.common.util.Constants;
+
+@SuppressWarnings("WeakerAccess")
+public class MMDContainer extends Container {
+    private final static double MAX_INTERACT_DISTANCE = 64.0;
```

Comment:
Possibly change this to take the player reach attribute into consideration?

Reply:
good idea

Reply:
it now takes under consideration both this max value and player reach attribute.